src/llmq/quorums.cpp

```diff
@@ -989,39 +989,48 @@ static void DataCleanupHelper(CDBWrapper& db, std::set<uint256> skip_list)
 
         db.WriteBatch(batch);
 
-        LogPrint(BCLog::LLMQ, "CQuorumManager::%d -- %s removed %d\n", __func__, prefix, count);
+        LogPrint(BCLog::LLMQ, "CQuorumManager::%s -- %s removed %d\n", __func__, prefix, count);
     }
 
     pcursor.reset();
-    db.CompactFull();
+
+    if (compact) {
+        // Avoid using this on regular cleanups, use on db migrations only
+        LogPrint(BCLog::LLMQ, "CQuorumManager::%s -- compact start\n", __func__);
+        db.CompactFull();
+        LogPrint(BCLog::LLMQ, "CQuorumManager::%s -- compact end\n", __func__);
+    }
 }
 
-void CQuorumManager::CleanupOldQuorumData(const CBlockIndex* pIndex) const
+void CQuorumManager::StartCleanupOldQuorumDataThread(const CBlockIndex* pIndex) const
 {
-    if (!fMasternodeMode || pIndex == nullptr || (pIndex->nHeight % 576 != 0)) {
+    if (!fMasternodeMode || pIndex == nullptr || (pIndex->nHeight % 576 != 58 /* no DKGs running or to be started soon */)) {
```

Comment:
yeah, so that's because LLMQ_60_75 dkgMiningWindowStart = 42 and then for most of quorums dkgInterval = 24 i.e. 576 + 42 is only 6 blocks till next DKGs (576 + 24*2) so I moved to to the next cycle (48) + dkgMiningWindowStart (10) = 58

Reply:
Maybe add this as a comment at least